### DISCUSSION TITLE
Chance for background colour change option in future versions?

### DISCUSSION THREAD

**Discussion by nima-amin** [ruffle-rs/ruffle#2001] [General]
Might there be a chance to implement an option to change the background colour of an swf file in future (maybe stable) releases of the desktop version?
As you can see in the attached screenshot the background of the black cat is a very dark grey (supposedly it's colour code #999999 or something close to this)- much darker than in the original website. So far I can still use SWF Player to play this file (unfortunately also this fine little programme relies on the IE ActiveX element to function, so will stop working in near future), in which the grey background is alreday by default a lighter grey (#CCCCCC), however you can also with a few mouseclicks change the background to whatever colour you like at the moment.

![Ruffle Maukie](https://user-images.githubusercontent.com/75982635/102472235-66bd9e00-4056-11eb-8aa1-3097f4630ff4.png)

![SWF Player Maukie](https://user-images.githubusercontent.com/75982635/102472931-1bf05600-4057-11eb-93df-91cabb6bbd82.png)

![SWF Player Hintegrund](https://user-images.githubusercontent.com/75982635/102472300-776e1400-4056-11eb-8510-5a848a3a97dd.png)

**Comment by ousia** [comment]
@nima-amin,

many thanks for your question.

I wonder whether your question might be related to an already reported issue (#1913).

In any case, if your SWF file comes from http://www.sundropcrystal.com/maukie/maukie2.swf, I cannot see the light gray you display.

Adobe Flash:

![flash-display](https://user-images.githubusercontent.com/203767/103232939-14288f80-493c-11eb-8dc4-cd447953aa29.png)

Ruffle:

![ruffle-display](https://user-images.githubusercontent.com/203767/103232750-94022a00-493b-11eb-98b7-a8326862826e.png)

Sorry if I’m missing something (you might have modified the SWF file), but with the file provided in the address above, Ruffle seems to be doing the right thing.

**Comment by ousia** [comment]
Analyzing the file provided at http://www.sundropcrystal.com/maukie/maukie2.swf, I get:

```
[009]         3 SETBACKGROUNDCOLOR (33/33/33)
```

`#333333` is exactly what Flash and Ruffle display.

@nima-amin, wouldn’t it be possible that SWF Player 2.6 could be misbehaving?

https://www.adobe.com/support/flashplayer/debug_downloads.html contains the official projector for Windows (and for macOS and Linux, as well).

Could you check your file with the Adobe Flash projector?

And if you modified http://www.sundropcrystal.com/maukie/maukie2.swf, could you upload your modified version?

Many thanks for your help.

**Comment by ousia** [comment]
@nima-amin, I tested that on both Linux and Windows computers, using latest version of the Adobe Flash projector.

Ruffle displays `#333333` as background color, the same that the Adobe Flash projector does in both Linux and Windows versions.

I there is something wrong with my explanation, I think it would be better to open an issue (to report a bug).

**Comment by n0samu** [comment]
Setting a custom background color in the desktop app is still unsupported, unfortunately. Now that we have a GUI, it would be a great thing for us to add!
